core/src/main/scala/chisel3/internal/MonoConnect.scala

```diff
@@ -178,6 +188,35 @@ private[chisel3] object MonoConnect {
       case (sink, source) => throw MismatchedException(sink.toString, source.toString)
     }
 
+  /** Check whether two [[Vec]]s can be bulk connected (<-) in FIRRTL. */
```

Comment:
can you add a bit more color here? What are the cases when they CANNOT be bulk-connected?